containerd-http-proxy/configure_http_proxy.yaml

```diff
@@ -0,0 +1,107 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: containerd-http-proxy
+spec:
+  selector:
+    matchLabels:
+      name: containerd-http-proxy
+  template:
+    metadata:
+      labels:
+        name: containerd-http-proxy
+    spec:
+      hostPID: true
+      containers:
+      - name: startup-script
+        image: gke.gcr.io/debian-base:bookworm-v1.0.0-gke.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          set -u
+
+          validate_proxy() {
```

Comment:
If the user doesn't specify HTTP_PROXY and HTTPS_PROXY in the configmap, the daemonset creation will fail because of missing env value. If they put empty string in configmap, validate_proxy() won't reject it.

Reply:
so if somebody configured an empty string - will everything work? I simply don't know how go will react on empty proxy

Reply:
If users have empty string in configmap, the pod will run successfully and http-proxy.conf will have something like`Environment="HTTP_PROXY="`. They will get image pull error if they want to pull new images because proxy is invalid. I think it'll be the same if they have any typos in proxy, but they can always fix the configmap and redeploy the daemonset.